### PR TITLE
Merged SLE-12-SP2

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

File: src/lib/network/network_autoyast.rb [hunk 1]
```diff
@@ -218,6 +218,15 @@ def ay_networking_section
       ay_profile["networking"]
     end
 
+    # Checks if the udev rule is valid for renaming a NIC
+    def valid_rename_udev_rule?(rule)
+      return false if rule["name"].nil? || rule["name"].empty?
+      return false if rule["rule"].nil? || rule["rule"].empty?
+      return false if rule["value"].nil? || rule["value"].empty?
+
+      true
```
Comment (jreidinger) [inline src/lib/network/network_autoyast.rb]: what about more dry check like:
```
["name", "rule", "value"].each do |key|
  return false if rule[key].nil? || rule[key].empty?
end

return true
```

File: src/lib/network/network_autoyast.rb [hunk 2]
```diff
@@ -228,7 +237,11 @@ def assign_udevs_to_devs(udev_rules)
       udev_rules.each do |rule|
         name_to = rule["name"]
         attr = rule["rule"]
-        key = rule["value"].downcase
+        key = rule["value"]
+
+        next if !valid_rename_udev_rule?(rule)
+        key.downcase!
```
Comment (jreidinger) [inline src/lib/network/network_autoyast.rb]: well, you modify now content if udev_rules, it is ok?

File: src/lib/network/network_autoyast.rb (hunk 1 above)
Comment (mchf) [inline src/lib/network/network_autoyast.rb]: matter of preference. My "simpler"  version is more readable for me.

File: src/lib/network/network_autoyast.rb (hunk 2 above)
Comment (mchf) [inline src/lib/network/network_autoyast.rb]: there should be no problem with that.

**Comment by mchf** [comment]
Thanks for review

**Pull request merged**